presto-main/src/main/java/com/facebook/presto/server/QueryStateInfo.java

```diff
@@ -74,13 +74,17 @@ public QueryStateInfo(
         this.progress = requireNonNull(progress, "progress is null");
     }
 
-    public static QueryStateInfo createQueryStateInfo(QueryInfo queryInfo, Optional<ResourceGroupInfo> group)
+    public static QueryStateInfo createQueuedQueryStateInfo(QueryInfo queryInfo, Optional<ResourceGroupId> group, Optional<List<ResourceGroupInfo>> pathToRoot)
     {
-        Optional<List<ResourceGroupInfo>> pathToRoot = group.map(ResourceGroupInfo::getPathToRoot);
-        return createQueryStateInfo(queryInfo, group.map(ResourceGroupInfo::getId), pathToRoot);
+        return createQueryStateInfo(queryInfo, group, pathToRoot);
     }
 
-    public static QueryStateInfo createQueryStateInfo(QueryInfo queryInfo, Optional<ResourceGroupId> groupId, Optional<List<ResourceGroupInfo>> pathToRoot)
+    public static QueryStateInfo createRunningQueryStateInfo(QueryInfo queryInfo, Optional<ResourceGroupId> group)
+    {
+        return createQueryStateInfo(queryInfo, group, Optional.empty());
+    }
+
+    private static QueryStateInfo createQueryStateInfo(QueryInfo queryInfo, Optional<ResourceGroupId> groupId, Optional<List<ResourceGroupInfo>> pathToRoot)
```

Comment:
Since you have two separate functions now for queued and running queries, should we move the specific logic for not queued queries (progress to the running query function? Also it's actually not necessarily running queries. It's everything that's "not queued" query.